cachito/workers/pkg_managers/npm.py

```diff
@@ -332,16 +332,20 @@ def get_package_and_deps(package_json_path, package_lock_path):
     return rv
 
 
-def resolve_npm(app_source_path, request):
+def resolve_npm(app_source_path, request, skip_deps=None):
     """
     Resolve and fetch npm dependencies for the given app source archive.
 
     :param str app_source_path: the full path to the application source code
     :param dict request: the Cachito request this is for
+    :param set skip_deps: a set of dependency identifiers to not download because they've already
+        been downloaded for this request
     :return: a dictionary that has the keys "deps" which is the list of dependencies,
-        "lock_file" which is the lock file if it was modified, "package" which is the
-        dictionary describing the main package, and "package.json" which is the package.json file if
-        it was modified.
+        "downloaded_deps" which is a set of the dependency identifiers of the dependencies that
+        were downloaded as part of this function's execution, "lock_file" which is the lock file if
+        it was modified, "lock_file_name" is the name of the lock file that was used, "package"
```

Comment:
Is it just the name of the lock file, or the full path to it?

Reply:
Just the name.